### PR TITLE
Send communicator messages to teachers from workspace front page

### LINKED ISSUE #3112: Send communicator messages to teachers from workspace front page
#### Details

 - There's a button for each teacher in workspace teachers list
 - Clicking the button shows a message dialog in the bottom of the page similar to the one in communicator, but with no recipient shown

### PR DESCRIPTION
**Pull request by pkukkon**
- resolves #3112

### CODE REVIEW

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 1]
```diff
@@ -892,63 +901,63 @@ sup {
...
+    clear: both;
+    font-size: 13px;
+    line-height: 1.2em;
+    margin: 0 -20px -10px;
+    padding: 5px 20px; 
+     
+    .producers-override {
+      margin: 10px 0;
+    
+      .material-producers-label {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 2]
```diff
@@ -892,63 +901,63 @@ sup {
...
+        color: #727272;
+        padding: 0 5px 0 0;
+        text-transform: uppercase;
+      }
+      
+      .material-producer {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 3]
```diff
@@ -892,63 +901,63 @@ sup {
...
-    .material-producer {
-      padding: 0;
-      color: #000;
-      text-transform: uppercase;
     }
+    
+    .license-override {
+      margin: 10px 0;
+    
+      .material-license-label {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 4]
```diff
@@ -892,63 +901,63 @@ sup {
...
-
-    .material-license {
-      padding: 0;
-      
-      a {
-        color: #09f;
+      .material-license {
+        padding: 0;
+        
+        a {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss (hunk 1 above)
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss (hunk 2 above)
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss (hunk 3 above)
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 5]
```diff
@@ -892,63 +901,63 @@ sup {
...
-      padding: 0 5px 0 0;
-      text-transform: uppercase;
-    }
-
-    .material-license {
-      padding: 0;
-      
-      a {
-        color: #09f;
+      .material-license {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss (hunk 4 above)
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 6]
```diff
@@ -1,767 +1,786 @@
...
+      }
+      
+      // Show Assignment's Evaluation
+      &.muikku-show-evaluation-button {
+        
+        &.failed {
+          background: $evaluation-failed-color;
+        }
+        
+        &.incomplete {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 7]
```diff
@@ -1,767 +1,786 @@
...
+          background: $evaluation-incomplete-color;
+        }
+        
+        &.passed {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss [hunk 8]
```diff
@@ -1,767 +1,786 @@
...
+    }
+    
+    .assignment-literal-data {
+      background: $evaluation-modal-assignment-evaluated-literal-background-color;
+      border-radius: 3px;
+      font-size: 14px;
+      line-height: 20px;
+      padding: 1px 10px;
+      
+      p {
```
Comment (jjlankinen) [inline muikku/src/main/webapp/resources/theme-muikku/sass/flex/_workspace-materials-defaults.scss]: At this point there's nothing we will do about this.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/workspace-frontpage.js [hunk 9]
```diff
@@ -2,9 +2,104 @@
...
+          staffMembers.sort(function(a, b) {
+            var an = a.lastName + ' ' + a.firstName;
+            var bn = b.lastName + ' ' + b.firstName;
+            return an < bn ? -1 : an == bn ? 0 : 1;
+          });
+          for (var i = 0; i < staffMembers.length; i++) {
+            var props = staffMembers[i].properties;
+            if (props['profile-vacation-start'] && props['profile-vacation-end']) {
+              var bd = moment(props['profile-vacation-start']).toDate();
+              bd.setHours(0);
```
Comment (ilmoeuro) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/workspace-frontpage.js]: We're using `moment`, which has built-in functionality for this:

```js
moment(...).startOf('day').isBefore(moment(...).startOf('day'))
```

**Pull request closed without merging**